backend/inspirehep/utils.py

```diff
@@ -45,3 +45,8 @@ def hash_data(data):
     if data:
         return hashlib.md5(data).hexdigest()
     raise ValueError("Data for hashing cannot be empty")
+
+
+def set_headers_for_record_caching_and_concurrency(response, record):
+    response.set_etag(str(record.revision_id), weak=True)
+    response.vary = "Content-Type,Accept,Accept-Encoding,Accept-Language"
```

Comment:
I don't think you need the `content-type`, it seems vary is used to add additional *request* headers to the cache key.